plugins/tabletools/dialogs/tableCell.js

```diff
@@ -8,7 +8,8 @@ CKEDITOR.dialog.add( 'cellProperties', function( editor ) {
 		langCell = langTable.cell,
 		langCommon = editor.lang.common,
 		validate = CKEDITOR.dialog.validate,
-		widthPattern = /^(\d+(?:\.\d+)?)(px|%)$/,
+		defaultLengthUnit = 'px';
+		lengthUnitPattern = /^(\d+(?:\.\d+)?)(px|%|mm|cm|pt|em|ex)$/,
```

Comment:
Your changes / new features are still not covered by unit tests. This is something I've requested in [my former comment](https://github.com/ckeditor/ckeditor-dev/pull/319#issuecomment-279025904).